pandas/core/series.py

```diff
@@ -431,13 +431,21 @@ def nonzero(self):
 
     def put(self, *args, **kwargs):
         """
-        return a ndarray with the values put
+        Applies the `put` methods to its `values` attribute
+        and returns the resulting ndarray. Please refer to
+        `numpy.ndarray.put` for full documentation.
 
         See also
         --------
         numpy.ndarray.put
         """
-        self._values.put(*args, **kwargs)
+        try:
+            self._values.put(*args, **kwargs)
+        except AttributeError:
+            klass = type(self._values).__name__
```

Comment:
no add a ``.put`` method to ``Categorical`` (and/or ``DatetimeIndex`` if needed). We always defer to these extension types, rather than trying to catch an error